File: aiagent7-service/src/main/java/com/shuangqi/aiagent7/common/BusinessException.java
package com.shuangqi.aiagent7.common;

import lombok.Getter;

//throw new BusinessException(400, "Invalid parameter");
//// 或
// throw new BusinessException("Business error message");
@Getter
public class BusinessException extends RuntimeException {
    private final Integer code;
    private final String message;

    public BusinessException(Integer code, String message) {
        super(message);
        this.code = code;
        this.message = message;
    }

    public BusinessException(String message) {
        this(500, message);
    }
}

File: langhuan-server/src/main/kotlin/com/langhuan/common/BusinessException.kt
package com.langhuan.common

//throw new BusinessException(400, "Invalid parameter");
//// 或
// throw new BusinessException("Business error message");
class BusinessException : RuntimeException {
    val code: Int
    override val message: String

    constructor(code: Int, message: String) : super(message) {
        this.code = code
        this.message = message
    }

    constructor(message: String) : this(500, message)
}
